src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -573,73 +645,400 @@ void ompl::geometric::RRTstar::getPlannerData(base::PlannerData &data) const
-            pruneScratchSpace_.toBePruned.push_back(candidate);
+    base::Cost costToCome;
+    if (useAdmissibleCostToCome_)
+    {
+        // Start with infinite cost
+        costToCome = opt_->infiniteCost();
+
+        //Find the min from each start
+        for (unsigned int i = 0u; i < startMotions_.size(); ++i)
+        {
+            costToCome = opt_->betterCost(costToCome, opt_->motionCost(startMotions_.at(i)->state, motion->state)); // lower-bounding cost from the start to the state
+        }
+    }
+    else
+    {
+        costToCome = motion->cost; // current cost from the state to the goal
+    }
+
+    const base::Cost costToGo = opt_->costToGo(motion->state, pdef_->getGoal().get()); // lower-bounding cost from the state to the goal
+    return opt_->combineCosts(costToCome, costToGo); // add the two costs
+}
+
+void ompl::geometric::RRTstar::setTreePruning(const bool prune)
+{
+    if (opt_->hasCostToGoHeuristic() == false)
+    {
+        OMPL_INFORM("%s: No cost-to-go heuristic set. Informed techniques will not work well.", getName().c_str());
+    }
+
+    useTreePruning_ = prune;
+
+    // If we're using prunedMeasure, we need to disable that
+    if (usePrunedMeasure_ == true && useTreePruning_ == false)
+    {
+        setPrunedMeasure(false);
+    }
+}
+
+void ompl::geometric::RRTstar::setPrunedMeasure(bool informedMeasure)
+{
+    if (opt_->hasCostToGoHeuristic() == false)
+    {
+        OMPL_INFORM("%s: No cost-to-go heuristic set. Informed techniques will not work well.", getName().c_str());
+    }
+
+    // This option only works with informed sampling
+    if (informedMeasure == true && (useInformedSampling_ == false || useTreePruning_ == false))
+    {
+        OMPL_ERROR("%s: InformedMeasure requires InformedSampling and TreePruning.", getName().c_str());
+    }
+
+    // Check if we're changed and update parameters if we have:
+    if (informedMeasure != usePrunedMeasure_)
+    {
+        // Store the setting
+        usePrunedMeasure_ = informedMeasure;
+
+        // Update the prunedMeasure_ appropriately
+        if (usePrunedMeasure_)
+        {
+            prunedMeasure_ = infSampler_->getInformedMeasure(prunedCost_);
+        }
         else
         {
-            pruneScratchSpace_.newTree.push_back(candidate);
-            pruneScratchSpace_.candidates.insert(pruneScratchSpace_.candidates.end(),
-                candidate->children.begin(), candidate->children.end());
+            prunedMeasure_ = si_->getSpaceMeasure();
+        }
+
+        // And either way, update the rewiring radius if necessary
+        if (useKNearest_ == false)
+        {
+            calculateRewiringLowerBounds();
         }
     }
+}
 
-    // To create the new nn takes one order of magnitude in time more than just checking how many
-    // states would be pruned. Therefore, only prune if it removes a significant amount of states.
-    if ((double)pruneScratchSpace_.newTree.size() / tree_size < pruneStatesThreshold_)
```

Comment:
GitHub has changed the diff it associates for this line, unfortunately burying the previous conversation about it. This is a comment to draw attention to that.